src/Compilers/CSharp/Test/Semantic/Semantics/RecordTests.cs

```diff
@@ -3639,6 +3649,710 @@ .maxstack  3
+                //     public B(ref B b) => throw null; // 1, not recognized as copy constructor
+                Diagnostic(ErrorCode.ERR_UnexpectedOrMissingConstructorInitializerInRecord, "B").WithLocation(3, 12)
+                );
+        }
+
+        [Fact, WorkItem(44902, "https://github.com/dotnet/roslyn/issues/44902")]
+        public void CopyCtor_BaseHasRefKind_WithThisInitializer()
+        {
+            var source =
+@"public record B(int i)
+{
+    public B(ref B b) : this(0) => throw null; // 1, not recognized as copy constructor
+}
+public record C(int j) : B(1)
+{
+    internal C(C c) : base(c)
+    {
+    }
+}
+";
+            var comp = CreateCompilation(source);
+            comp.VerifyDiagnostics();
+
+            var actualMembers = comp.GetMember<NamedTypeSymbol>("B").GetMembers().Where(m => m.Name == ".ctor").ToTestDisplayStrings();
+            var expectedMembers = new[]
+            {
+                "B..ctor(System.Int32 i)",
+                "B..ctor(ref B b)",
+                "B..ctor(B )"
+            };
+            AssertEx.Equal(expectedMembers, actualMembers);
+        }
+
+        [Fact, WorkItem(44902, "https://github.com/dotnet/roslyn/issues/44902")]
+        public void CopyCtor_WithPrivateField()
+        {
+            var source =
+@"public record B(object N1, object N2)
+{
+    private int field1 = 100;
+    public int GetField1() => field1;
+}
+public record C(object P1, object P2) : B(3, 4)
+{
+    private int field2 = 200;
+    public int GetField2() => field2;
+
+    static void Main()
+    {
+        var c1 = new C(1, 2);
+        var c2 = new C(c1);
+        System.Console.Write((c2.P1, c2.P2, c2.N1, c2.N2, c2.GetField1(), c2.GetField2()));
+    }
+}";
+            var comp = CreateCompilation(new[] { source, IsExternalInitTypeDefinition }, parseOptions: TestOptions.RegularPreview, options: TestOptions.ReleaseExe);
+            comp.VerifyDiagnostics();
+
+            var verifier = CompileAndVerify(comp, expectedOutput: "(1, 2, 3, 4, 100, 200)", verify: ExecutionConditionUtil.IsCoreClr ? Verification.Skipped : Verification.Fails);
+            verifier.VerifyIL("C..ctor(C)", @"
+{
+  // Code size       44 (0x2c)
+  .maxstack  2
+  IL_0000:  ldarg.0
+  IL_0001:  ldarg.1
+  IL_0002:  call       ""B..ctor(B)""
+  IL_0007:  ldarg.0
+  IL_0008:  ldarg.1
+  IL_0009:  ldfld      ""object C.<P1>k__BackingField""
+  IL_000e:  stfld      ""object C.<P1>k__BackingField""
+  IL_0013:  ldarg.0
+  IL_0014:  ldarg.1
+  IL_0015:  ldfld      ""object C.<P2>k__BackingField""
+  IL_001a:  stfld      ""object C.<P2>k__BackingField""
+  IL_001f:  ldarg.0
+  IL_0020:  ldarg.1
+  IL_0021:  ldfld      ""int C.field2""
+  IL_0026:  stfld      ""int C.field2""
+  IL_002b:  ret
+}");
+        }
+
+        [Fact, WorkItem(44902, "https://github.com/dotnet/roslyn/issues/44902")]
+        public void CopyCtor_MissingInMetadata()
+        {
+            // IL for `public record B { }`
```

Comment:
Can we define this in C# (as a `class`) rather than IL? Same question for subsequent test.

Reply:
No, we cannot write the `<>Clone` method in C#

---
In reply to: [438829672](https://github.com/dotnet/roslyn/pull/45022#discussion_r438829672) [](ancestors = 438829672)